core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/WorkflowWorker.scala

```diff
@@ -0,0 +1,289 @@
+    }
+  }
+
+  def onResumeTuple(faultedTuple: FaultedTuple): Unit = {
+    if (!faultedTuple.isInput) {
+      batchProducer.passTupleToDownstream(faultedTuple.tuple)
+    } else {
+      dataProcessor.prependElement(InputTuple(faultedTuple.tuple))
+    }
+  }
+
+  def onModifyTuple(faultedTuple: FaultedTuple): Unit = {
+    if (!faultedTuple.isInput) {
+      batchProducer.passTupleToDownstream(faultedTuple.tuple)
+    } else {
+      dataProcessor.prependElement(InputTuple(faultedTuple.tuple))
+    }
+  }
+
+  def getInputRowCount(): Long = {
+    val (inputCount, _) = dataProcessor.collectStatistics()
+    inputCount
+  }
+
+  def getOutputRowCount(): Long = {
+    val (_, outputCount) = dataProcessor.collectStatistics()
+    outputCount
+  }
+
+  def getResultTuples(): mutable.MutableList[ITuple] = {
+    this.operator match {
+      case processor: ITupleSinkOperatorExecutor =>
+        mutable.MutableList(processor.getResultTuples(): _*)
+      case _ =>
+        mutable.MutableList()
+    }
+  }
+
+  override def receive: Receive = receiveAndProcessMessages
+
+  def receiveAndProcessMessages: Receive = {
+    routeActorRefRelatedMessages orElse
+      processControlMessages orElse
+      oldControlMessageHandler orElse
+      receiveDataMessages orElse {
+      case other =>
+        logger.info(s"unhandled message: $other")
+    }
+  }
+
+  def oldControlMessageHandlingLogic: Receive = {
+    case ExecutionCompleted() =>
+      isCompleted = true
+      workerStateManager.transitTo(Completed)
+      reportState()
+    case ReturnPayload(_, v: ExecutionPaused) =>
+      workerStateManager.transitTo(Paused)
+      reportState()
+    case UpdateInputLinking(identifier, inputNum) =>
+      logger.info(s"received register input for ${this.identifier}")
+      sender ! Ack
+      tupleProducer.registerInput(identifier, inputNum)
+    case LocalBreakpointTriggered() =>
+      dataProcessor.breakpoints.foreach { brk =>
+        if (brk.isTriggered)
+          dataProcessor.unhandledFaultedTuples(brk.triggeredTupleId) =
+            new FaultedTuple(brk.triggeredTuple, brk.triggeredTupleId, brk.isInput)
+      }
+      context.parent ! ReportState(WorkerState.LocalBreakpointTriggered)
+    case other =>
+  }
+
+  def getOldWorkerState: WorkerState.Value = {
+    workerStateManager.getCurrentState match {
+      case UnInitialized =>
+        WorkerState.Uninitialized
+      case Ready =>
+        WorkerState.Ready
+      case Running =>
+        WorkerState.Running
+      case Pausing =>
+        WorkerState.Pausing
+      case Paused =>
+        WorkerState.Paused
+      case Completed =>
+        WorkerState.Completed
+      case Recovering =>
+        WorkerState.Running
+    }
+  }
+
+  def reportState(): Unit = context.parent ! ReportState(getOldWorkerState)
+
+  def oldControlMessageHandler: Receive = {
+    case Start =>
+      sender ! Ack
+      if (workerStateManager.getCurrentState != Ready) {
+        logger.info(
```

Comment:
Same comment here. Shouldnt this be an error?

Reply:
done